include/zephyr/drivers/comparator/nrf_comp.h

```diff
@@ -0,0 +1,160 @@
+	/** AIN7 external input */
+	COMP_NRF_COMP_INPUT_7,
+	/** VDD / 2 */
+	COMP_NRF_COMP_VDD_DIV2,
+	/** VDD / 5 */
+	COMP_NRF_COMP_VDDH_DIV5,
+};
+
+/** Negative external input selection */
+enum comp_nrf_comp_ext_ref {
+	/** AIN0 external input */
+	COMP_NRF_COMP_EXT_REF_0,
+	/** AIN1 external input */
+	COMP_NRF_COMP_EXT_REF_1,
+	/** AIN2 external input */
+	COMP_NRF_COMP_EXT_REF_2,
+	/** AIN3 external input */
+	COMP_NRF_COMP_EXT_REF_3,
+	/** AIN4 external input */
+	COMP_NRF_COMP_EXT_REF_4,
+	/** AIN5 external input */
+	COMP_NRF_COMP_EXT_REF_5,
+	/** AIN6 external input */
+	COMP_NRF_COMP_EXT_REF_6,
+	/** AIN7 external input */
+	COMP_NRF_COMP_EXT_REF_7,
+};
+
+/** Negative internal input selection */
+enum comp_nrf_comp_ref {
+	/** Internal 1.2V reference */
+	COMP_NRF_COMP_REF_INT_1V2,
+	/** Internal 1.8V reference */
+	COMP_NRF_COMP_REF_INT_1V8,
+	/** Internal 2.4V reference */
+	COMP_NRF_COMP_REF_INT_2V4,
+	/** AVDD 1.8V reference */
+	COMP_NRF_COMP_REF_AVDDAO1V8,
+	/** VDD reference */
+	COMP_NRF_COMP_REF_VDD,
+	/** Use negative external reference */
+	COMP_NRF_COMP_REF_AREF,
+};
+
+/** Speed mode selection */
+enum comp_nrf_comp_speed_mode {
+	/** Low-power mode */
+	COMP_NRF_COMP_SP_MODE_LOW,
+	/** Normal mode */
+	COMP_NRF_COMP_SP_MODE_NORMAL,
+	/** High-speed mode */
+	COMP_NRF_COMP_SP_MODE_HIGH,
+};
+
+/** Hysteresis configuration */
+enum comp_nrf_comp_hyst {
+	/** No hysteresis */
+	COMP_NRF_COMP_HYST_NO_HYST,
+	/** 40mV hysteresis */
+	COMP_NRF_COMP_HYST_40MV,
+	/** 50mV hysteresis */
+	COMP_NRF_COMP_HYST_50MV,
+};
+
+/** Current source configuration */
+enum comp_nrf_comp_isource {
+	/** Current source disabled */
+	COMP_NRF_COMP_ISOURCE_OFF,
+	/** 2.5uA current source enabled */
+	COMP_NRF_COMP_ISOURCE_IEN_2UA5,
+	/** 5uA current source enabled */
+	COMP_NRF_COMP_ISOURCE_IEN_5UA,
+	/** 10uA current source enabled */
+	COMP_NRF_COMP_ISOURCE_IEN_10UA,
+};
+
+/**
+ * @brief Configure comparator in single-ended mode using internal reference
+ *
+ * @param dev Comparator device instance
+ * @param input Positive input selection
+ * @param speed_mode Speed mode selection
+ * @param isource Current source configuration
+ * @param ext_ref Negative external input selection
+ * @param ref Negative input selection
+ * @param th_down Hysteresis down threshold configuration
+ * @param th_up Hysteresis up threshold configuration
+ *
+ * @note ext_ref is only used if ref == COMP_NRF_COMP_REF_AREF
+ * @note Hysteresis down in volts = ((th_down + 1) / 64) * ref
+ * @note Hysteresis up in volts = ((th_up + 1) / 64) * ref
+ *
+ * @retval 0 if successful
+ * @retval negative errno-code otherwise
+ */
+int comp_nrf_comp_configure_se(const struct device *dev,
+			       enum comp_nrf_comp_input input,
+			       enum comp_nrf_comp_speed_mode speed_mode,
+			       enum comp_nrf_comp_isource isource,
+			       enum comp_nrf_comp_ext_ref ext_ref,
+			       enum comp_nrf_comp_ref ref,
+			       uint8_t th_down,
+			       uint8_t th_up);
```

Comment:
I think it would be better to use a structure instead of this bunch of parameters. This way, users would not be forced to provide a value for parameters they don't won't or cannot use (like `ext_ref` when an internal reference is used or `isource` on some SoCs). A structure can also be extended with a new field (if such need arises) without breaking the API.

Reply:
I will change it to be a structure :)

Reply:
Done